tests/run/cygdb_venv.srctree

```diff
@@ -0,0 +1,81 @@
+# mode: run
+# tag: py3only, gdb
+# ticket: 1961
+
+PYTHON -m venv venv
+PYTHON test_runner.py
+
+######## test_runner.py ########
+
+import sys, os
+import subprocess
+
+venv_args = 'source venv/bin/activate'
+build_args = 'python setup.py build_ext -i'
+dummy_install_args = 'python -m pip install dummy_module_dir/'
+cygdb_run_args = 'python -c "from Cython.Debugger import Cygdb as cygdb; cygdb.main()" . -- -ex "run test_inside_gdb.py" -ex quit python'
+
+command = ' && '.join([
+    venv_args,
+    build_args,
+    dummy_install_args,
+    cygdb_run_args,
+])
+
+p = subprocess.Popen(['bash', '-c', command], stderr=subprocess.PIPE, stdout=subprocess.PIPE, text=True)
+_out, _err = p.communicate()
+res = p.returncode
```

Comment:
This is Py3-only, you can use `subprocess.check_call()`. You don't even have to collect the output, just let `runtests.py` do it for you. Run separate commands to avoid the dependency on a shell.

Reply:
Thanks for suggestion, I will do that